frame/elections-phragmen/src/migrations/v5.rs

```diff
@@ -0,0 +1,65 @@
+use super::super::*;
+//  use frame_support::traits::Get;
+
+/// Migrate the locks and vote stake on accounts that have more than their free balance locked.
+///
+/// This migration addresses a bug were a voter could lock up to their reserved balance + free
+/// balance. Since locks are only designed to operate on free balance, this put those affected in a
+/// situation where they could increase their free balance but still not be able to use there funds
+/// because it is less than the lock.
+pub fn migrate<T: Config>() -> Weight {
+	let mut weight = 0;
+
+	for (who, mut voter) in Voting::<T>::iter() {
+		let free_balance = T::Currency::free_balance(&who);
+		let locked = voter.stake;
+
+		weight = T::DbWeight::get().reads(2);
+
+		if locked > free_balance {
+			voter.stake = free_balance;
+			Voting::<T>::insert(&who, voter);
+
+			let pallet_id = T::PalletId::get();
+			T::Currency::remove_lock(pallet_id, &who);
+			T::Currency::set_lock(pallet_id, &who, free_balance, WithdrawReasons::all());
```

Comment:
```suggestion
			T::Currency::set_lock(pallet_id, &who, free_balance, WithdrawReasons::all());
```

I think `set_lock` will just update the lock to the new amount, so removing is not needed.

Reply:
Ah yup you are correct - the docs say

```
 /// If the lock `id` already exists, this will update it.
```

Reply:
4f40a1f